airbyte-cdk/python/airbyte_cdk/entrypoint.py

```diff
@@ -139,12 +140,28 @@ def check(self, source_spec: ConnectorSpecification, config: TConfig) -> Iterabl
             self.validate_connection(source_spec, config)
         except AirbyteTracedException as traced_exc:
             connection_status = traced_exc.as_connection_status_message()
+            # The platform uses the exit code to surface unexpected failures so we raise the exception if the failure type is system_error
+            # If the failure is not exceptional, we'll emit a failed connection status message and return
+            if traced_exc.failure_type == FailureType.system_error:
+                raise traced_exc
             if connection_status:
                 yield from self._emit_queued_messages(self.source)
                 yield connection_status
                 return
 
-        check_result = self.source.check(self.logger, config)
+        try:
+            check_result = self.source.check(self.logger, config)
+        except AirbyteTracedException as traced_exc:
+            yield traced_exc.as_airbyte_message()
+            # The platform uses the exit code to surface unexpected failures so we raise the exception if the failure type is system_error
+            # If the failure is not exceptional, we'll emit a failed connection status message and return
+            if traced_exc.failure_type == FailureType.system_error:
```

Comment:
What about transient_error? Shouldn't this also return exit code -1?

Reply:
probably, yes. updated to != config error